File: server/src/main/java/org/tokio/teste/arthur/utils/LoginResponseDTO.java
package org.tokio.teste.arthur.utils;

import lombok.Data;

import java.io.Serializable;
import java.util.Date;

@Data
public class LoginResponseDTO implements Serializable {
    Boolean ok = true;
    String token;
    String name;
    String email;
    String nickname;
    String language = "pt";
    Boolean dark = true;
    String kind;
    Date loginAt = new Date();
}

File: web/src/app/share/models/utility/LoginResponseDTO.ts
import {User} from "@app/share/models/User";

export class LoginResponseDTO {
    ok: boolean = false;
    name: string;
    nickname: string;
    email: string;
    language: string;
    dark: boolean;
    token: string;
    kind: 'member' | 'admin';
    loginAt = new Date();

    toUser(): User {
        let ret = new User()
        ret.name = this.name
        ret.nickname = this.nickname
        ret.email = this.email
        ret.language = this.language
        ret.darkTheme = this.dark
        ret.kind = this.kind
        return ret
    }
}
